### PR TITLE
Fix mines

### LINKED ISSUE #10110: Can't walk over mines
(WEB REPORT BY: zealotron REMOTE: 172.93.109.202:7777)

> Revision (Should be above if you're viewing this from ingame!)
> fbd3015fda4e65f37b5a63b77d1e5f075680290f 
> 
> General description of the issue
> You can't walk over the N2/Plasma mines that are on the, uhh, sand cult vault away mission.
> 
> What you expected to happen
> To be able to walk over a mine, maybe have my legs blow up.
> 
> What actually happened
> I could not walk over the mine, it blocked off an area.
> 
> Steps to reproduce if possible
> try to walk over a mine.

### PR DESCRIPTION
**Pull request by clusterfack**
Fixes #10110
Also fixes kick mines not working

### CODE REVIEW

**Comment by clusterfack** [comment]
I removed the nonpolymorphic nonsense and moved this manually call nonsense into a single trigger proc for each type. And changed qdel(client) into del(client) so it would actually do something.

File: code/game/objects/effects/mines.dm [hunk 1]
```diff
@@ -11,42 +11,56 @@
...
-		M.Stun(30)
+	AM.radiation += 50
+	randmutb(AM)
+	domutcheck(AM,null)
+	qdel(src)
+
+/obj/effect/mine/plasma
+	name = "Plasma Mine"
+
+/obj/effect/mine/plasma/proc/trigger(AM)
```
Comment (Iamgoofball) [inline code/game/objects/effects/mines.dm]: why are you redefining trigger as a new proc on every mine

File: code/game/objects/effects/mines.dm [hunk 2]
```diff
@@ -11,42 +11,56 @@
...
+/obj/effect/mine/kick
+	name = "Kick Mine"
+
+/obj/effect/mine/kick/proc/trigger(mob/AM)
 	var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread
 	s.set_up(3, 1, src)
 	s.start()
-	spawn(0)
-		qdel(src)
+	del(AM.client)
```
Comment (ComicIronic) [inline code/game/objects/effects/mines.dm:57]: > del

Is this safe

**Comment by ComicIronic** [comment]
Now that's OOP :ok_hand:

File: code/game/objects/effects/mines.dm (hunk 2 above)
Comment (clusterfack) [inline code/game/objects/effects/mines.dm]: Not only is it safe, it's necessary!

File: code/game/objects/effects/mines.dm (hunk 1 above)
Comment (clusterfack) [inline code/game/objects/effects/mines.dm]: Because each mine does something different when you trigger it

Comment (clusterfack) [inline code/game/objects/effects/mines.dm]: Ah nvm I see what you mean I'm fixing that

Comment (ComicIronic) [inline code/game/objects/effects/mines.dm]: Is this legitimate DM or was it causing compiler errors

Comment (clusterfack) [inline code/game/objects/effects/mines.dm]: That was what caused the compiler error, it was a leftover from when I converted the individual procs under the same name trigger.

**Pull request merged**